osu.Game/Screens/Edit/Components/Timelines/Summary/Parts/TimelinePart.cs

```diff
@@ -57,15 +57,13 @@ private void load(IBindable<WorkingBeatmap> beatmap, EditorClock clock)
 
         private void updateRelativeChildSize()
         {
-            // the track may not be loaded completely (only has a length once it is).
+            // If the track is not loaded, assign a default sane length otherwise relative positioning becomes meaningless.
+            double trackLength = beatmap.Value.Track.IsLoaded ? beatmap.Value.Track.Length : 60000;
+            content.RelativeChildSize = new Vector2((float)Math.Max(1, trackLength), 1);
+
+            // The track may not be loaded completely (only has a length once it is).
             if (!beatmap.Value.Track.IsLoaded)
-            {
-                content.RelativeChildSize = Vector2.One;
                 Schedule(updateRelativeChildSize);
-                return;
-            }
-
-            content.RelativeChildSize = new Vector2((float)Math.Max(1, beatmap.Value.Track.Length), 1);
```

Comment:
The test still passes without this change, but the controlpoint is some 3 million pixels offscreen since blueprints are assigned relative-x values of e.g. 2000 ms, which results in `2000 * screen_width` final x coordinate. I felt it was wrong to leave it in this state.